Toggle password added

niksm7 opened the pull request:
# Related Issue
Fixes: #210 

# Proposed Changes
- Added an Eye icon besides the password input field that helps in toggling the visibility of password and changes the icon to slashed eye when the visibility of the password is on.
- Responsiveness of the page is maintained.


# Screenshots

Original             |  Updated
:-------------------------:|:-------------------------:
<img width="806" alt="Screenshot 2021-04-01 at 1 41 27 PM" src="https://user-images.githubusercontent.com/55614782/113264006-1a8a2700-92f0-11eb-88ca-e3c728048cb0.png"> | <img width="806" alt="Screenshot 2021-04-01 at 1 41 55 PM" src="https://user-images.githubusercontent.com/55614782/113264036-22e26200-92f0-11eb-8e93-c00752f4ebaf.png">

tushhr commented on templates/accounts/login_signup.html:
```diff
@@ -90,6 +101,29 @@ <h1>Hello, Friend!</h1>
     </div>
 </div>
 
+<script>
```
Write this script in the `accounts/style.js` in the static folder as that's the corresponding js file for this page.

tushhr commented on templates/accounts/login_signup.html:
```diff
@@ -69,7 +70,17 @@ <h1>Sign in</h1>
-            <input type="password" name="password" placeholder="Password" required />
+
+            <!-- Div to hold the elements in single line -->
+            <div style="display: table;width: 100%;margin-left: -12px;">
```
Don't write so many inline styles. You can create a new class and specify the properties there.

tushhr commented on templates/accounts/login_signup.html:
```diff
@@ -69,7 +70,17 @@ <h1>Sign in</h1>
+                <input type="password" name="password" id='password' style="display: inline-block" placeholder="Password" required>
+
+                <!-- Eye icon to toggle -->
+                <i class="far fa-eye" id="togglePassword" style="margin-left: -30px; cursor: pointer;"></i>
```
Could you add some sign or something on hover which will inform the user that this icon is to show/hide the password.

garg3133 replied:
A tooltip should be fine.

tushhr reviewed (changes requested):
Hey, @niksm7 just a few small changes. Rest everything looks goods!

niksm7 commented:
Will do all the required changes and push the code @garg3133 @tushhr

garg3133 commented:
Also, as we need only two icons for this page, we should avoid using the complete font-awesome library and download thousand of icons for our users. Instead, we should only import those two icons and use them.

niksm7 commented:
> Also, as we need only two icons for this page, we should avoid using the complete font-awesome library and download thousand of icons for our users. Instead, we should only import those two icons and use them.

Yes will try to optimize it as much as possible

niksm7 commented:
Changes made:
- Shifted the script to style.js
- Removed inline css and added external css to style.css
- Added tooltip on hover about show/hide password
- Optimised the code by excluding complete loading of font-awesome icons and replacing it with 2 svg icons.

Please review the changes @garg3133 @tushhr

garg3133 commented on static/accounts/style.css:
```diff
@@ -315,4 +315,20 @@ input {
+}
+
+#password{
+	display: inline-block;
```
I don't see the need for this property as all input fields are already `inline-block` by default.

garg3133 commented on static/accounts/style.css:
```diff
@@ -315,4 +315,20 @@ input {
+#password_div{
+	display: table;
+	width: 100%;
+	margin-left: -12px;
```
Instead of using hard-coded negative margins, you can just change the `text-align` property of table to `left`.

You can visualize things by putting a border around the elements.

garg3133 reviewed (commented):
@niksm7 The PR overall looks good now, but that eye staring at you look a little scary. Due to its black colour, it is getting highlighted, so maybe we can change its colour to something of lighter shade (like the text of placeholder).

@tushhr What do you think?

niksm7 commented:
- Changed color of the eye icon to lighter shade
- Removed extra css "inline-block"
- Replaced hard coded margin

@garg3133

garg3133 commented:
@niksm7 Please also attach a screenshot whenever you make some changes in the UI. It saves a lot of maintainer's time while giving feedback.

niksm7 commented:
<img width="925" alt="Screenshot 2021-04-03 at 4 11 23 PM" src="https://user-images.githubusercontent.com/55614782/113476156-4693dc80-9497-11eb-8a44-5bbeefcf7aeb.png">

Sorry for the inconvenience caused @garg3133

garg3133 commented:
Looks good @niksm7! Merging it into master 🎉 

Thank you for your contribution 🚀
